### ISSUE TITLE
'sharedApplication' is unavailable: not available on iOS (App Extension) - Use view controller based solutions where appropriate instead.

### ISSUE THREAD

**Issue by haoduynguyen** [invertase/react-native-firebase#5543]
https://rnfirebase.io/messaging/ios-notification-images

`target 'ImageNotification' do
  pod 'Firebase/Messaging'
end
`
After completing the steps to run the application, this error is reported (xcode 12.5.1, firebase messing 8.4.0)

**Comment by mikehardy** [comment]
Hi there, you have completely ignored the issue template, we close issues like that.

**Comment by rchavik** [comment]
Hi,

I understand the reason at the original issue did not follow the template, but this was a valid issue IMO.

Most solutions I came across, recommend adding/changing `ios/Podfile` somewhere along these lines:

```ruby
  target.build_configurations.each do |config|
    config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'NO'
  end
```

It does not work for me with `react-native` 0.64.3.

However, applying similar approach in `node_modules/react-native/scripts/react_native_pods.rb` worked.

Here is the `patches/react-native+0.64.3.patch` generated by `patch-package`:

```diff
diff --git a/node_modules/react-native/scripts/.packager.env b/node_modules/react-native/scripts/.packager.env
new file mode 100644
index 0000000..361f5fb
--- /dev/null
+++ b/node_modules/react-native/scripts/.packager.env
@@ -0,0 +1 @@
+export RCT_METRO_PORT=8081
diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index db9a34a..d3c1b5e 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -144,6 +144,12 @@ def react_native_post_install(installer)
     flipper_post_install(installer)
   end
 
+  installer.pods_project.targets.each do |target|
+    target.build_configurations.each do |config|
+      config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'NO'
+    end
+  end
+
   ## Fix for RTC-Folly on iOS 14.5 - makes Hermes work again
   find_and_replace(
     "#{pods_prefix}/RCT-Folly/folly/synchronization/DistributedMutex-inl.h",
```

No idea what changed .packager.env though. I left it as is.

So far, using the above, I managed to run via xcode.

Additionally,  I also came across an issue where the link process complained with `Undefined symbol: _OBJC_CLASS_$_RCTInspectorDevServerHelper`.

The solution was to move the Notification service target inside the main target, similar to this pseudo `ios/Podfile` (change $foo and $yourNotificationService accordingly):
```ruby

target '$foo' do

  # default react-native setup here

  target '$yourNotificationService' do
    pod 'Firebase/Messaging', '8.8.0'
  end

done
```
